pkg/expr/nodes.go

```diff
@@ -254,10 +254,13 @@ func executeDSNodesGrouped(ctx context.Context, now time.Time, vars mathexp.Vars
 		uid   string // in theory I think this all I need for the key, but rather be safe
```

Comment:
I think the comment on the enclosing function needs to be updated.

I'm not following the logic. We group now if the same data source instance and the same time range? If this is correct, then when data sources require grouping to work, won't this break the grouping (and functionality) if different time ranges are used in those data sources as well?

If that is all correct, maybe the path forward is to opt-in by data source type?

Reply:
> Opt-in is a less hacky solution though

I think so. When I added the feature - I didn't consider doing a mixed way however (some being grouped, and other not), so we will need to make sure that doesn't introduce any odd situations - but I think it will be easier to consider that once a draft exists.

Reply:
Sounds good, I'll close this then!

Reply:
I created an issue for supporting this in CloudWatch here: https://github.com/grafana/grafana/issues/85495